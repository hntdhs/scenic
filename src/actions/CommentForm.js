import React, { useState, useContext } from "react";
import { useParams } from "react-router-dom";
import BywayApi from "../api/api";
import UserContext from "../auth/UserContext";
import LimitedTextArea from "../common/LimitedTextArea";


function CommentForm({id, onAdd}) {
    const { byway } = useParams();
    const { currentUser } = useContext(UserContext);

    const [comment, setComment] = useState("");

    async function handleSubmit(evt) {
        const response = await BywayApi.makeComment(id, comment);
        setComment('');
        onAdd(response)
        // getCommentsByByway(byway);
    }

    return (
        <div>
            {/* <textarea onChange={e => setComment(e.target.value)} ></textarea> */}
            <LimitedTextArea domId="comment_form" value={comment} limit={500} onChange={setComment}/>
            <button id="comment_submit" onClick={handleSubmit} class="btn btn-outline-success btn-lg">
                Submit
            </button>
        </div>
    )
}

export default CommentForm;