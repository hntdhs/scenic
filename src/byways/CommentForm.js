import React, { useState, useContext } from "react";
import { useParams } from "react-router-dom";
import BywayApi from "../api/api";
import LoadingSpinner from "../common/LoadingSpinner"; 
import UserContext from "../auth/UserContext";

// don't forget link to edit, plus a ternary operator that displays a 'you haven't created a profile' message if there's nothing to display - was I going to make a table for profile data as opposed to user data?
// call api for user info, should be same as jobly

function CommentForm({name}) {
    const { byway } = useParams();
    // byway I think since that's what's in the URL
    const { currentUser } = useContext(UserContext);

    const [comment, setComment] = useState(false);
    // I think state is necessary because at first there is no comment, then comment exists

    async function handleSubmit(evt) {
        await BywayApi.makeComment(name, comment);
        // setComment('');
        setComment(comment);
    }

    return (
        <div>
            <textarea onChange={e => setComment(e.target.value)} ></textarea>
            <button onClick={handleSubmit}>
                Submit
            </button>
        </div>
    )
}

export default CommentForm;