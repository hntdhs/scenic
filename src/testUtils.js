import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};
// const
let UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
    )


// const UserProvider =
//     ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
//     <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
//       {children}
//     </UserContext.Provider>
// );

export { UserProvider };
