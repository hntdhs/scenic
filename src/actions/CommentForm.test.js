import React from "react";
import { render } from "@testing-library/react";
import CommentForm from "./CommentForm";
import { ToastProvider } from 'react-toast-notifications';
import { MemoryRouter } from "react-router-dom";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
    const { asFragment } = render(
    <ToastProvider>
      <MemoryRouter>
        <UserProvider>
        <CommentForm byway={'Great River Road'}/>
        </UserProvider>
        </MemoryRouter>
        </ToastProvider>);
    expect(asFragment()).toMatchSnapshot();
  });