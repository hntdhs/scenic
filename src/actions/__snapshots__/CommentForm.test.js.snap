// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <textarea />
      <p>
        Maximum length: 500
        <br />
        0/500
      </p>
    </div>
    <button>
      Submit
    </button>
  </div>
</DocumentFragment>
`;
